### DISCUSSION TITLE
Adding alpine.js as a module throws the following warning

### DISCUSSION THREAD

**Discussion by ineptian** [tailpress/tailpress#232] [Q&A]
https://alpinejs.dev/essentials/installation#as-a-module

```Alpine Warning: Unable to initialize. Trying to load Alpine before `<body>` is available. Did you forget to add `defer` in Alpine's `<script>` tag?```

Normally adding alpine this way doesnt throw errors but I am not exactly sure how everything is working behind the scenes in tailpress. Any suggestions? Thanks!

**Comment by weber-s** [comment]
Hello,

Simply wrap the Alpine.start() in a DOMContentLoaded:

```js
import Alpine from "alpinejs";

window.Alpine = Alpine;

document.addEventListener("DOMContentLoaded", () => {
    Alpine.start();
});
```

otherwise the JS is executed before the DOM is... loaded. :P

**Reply by ineptian** [reply]
I was actually able to get this working the normal way but I cant remember what I did. Could have just been user error.

**Comment by zerdotre** [comment]
You can also enqueue the cdn min. file in functions.php

**Reply by jeffreyvr** [reply]
Yes, and since a while with `wp_enqueue_script ` you can pass the `strategy` as part of the `args` array and set it to `defer`. 

https://developer.wordpress.org/reference/functions/wp_enqueue_script/
